frontend/src/components/ExpenseList.js

```diff
@@ -4,20 +4,62 @@ import { FaSave } from "react-icons/fa";
 import "../styles/ExpenseList.css";
 
 class ExpenseList extends React.Component {
+  state = {
+    userexpense: { categoryName: "", amount: "" },
+  };
+
+  onChange = (e) => {
+    // this changes the state of userexpenses amount.
+    this.setState((prevState) => {
+      let userexpense = Object.assign({}, prevState.userexpense);
+      userexpense[e.target.name] = e.target.value;
+      return { userexpense };
+    });
+  };
+
+  onSubmit = (e) => {
+    e.preventDefault();
+    if (!e.target.checkValidity()) {
+      return;
+    } else {
+      //this function is defined at expensepage and its updates total expenses
+      this.props.updateTotal(this.state.userexpense.amount);
+
+      //Clears the fields after taking input
+      e.target.reset();
+    }
+  };
```

Comment:
The function structure seems a bit messy, we could do just e.g.
``` JS
  onSubmit = (e) => {
    e.preventDefault();
    if (e.target.checkValidity())
      this.props.updateTotal(e.target[1].value) && e.target.reset();
  };
```

This way the component state and onChange function would not even be needed.

Reply:
Cleaned the on Submit function, but dint remove on Change, as I think, we will be needing this in adding transaction history.

Reply:
Tried doing it but for some reason it broke the application. Have no ideas, why. For now, making it as 
e.preventDefault();
    if (e.target.checkValidity()) {
      this.props.updateTotal(this.state.userexpense.amount);
      e.target.reset();
    }